lib/ansible/modules/network/vyos/vyos_static_routes.py

```diff
@@ -0,0 +1,719 @@
+
+
+# Using replaced
+#
+# Before state:
+# -------------
+#
+# vyos@vyos:~$ show configuration commands| grep static
+# set protocols static route 192.0.0.0/24 'blackhole'
+# set protocols static route 192.0.0.0/24 next-hop '192.11.11.11'
+# set protocols static route 192.0.0.0/24 next-hop '192.11.11.12'
+# set protocols static route6 2001:db8::0/32 blackhole distance '2'
+# set protocols static route6 2001:db8::0/32 next-hop '2001:db8::1'
+# set protocols static route6 2001:db8::0/32 next-hop '2001:db8::5'
+#
+- name: Replace device configurations of listed static routes with provided configurations
+  vyos_static_routes:
+    config:
+     - address_families:
+       - afi: 'ipv4'
+         routes:
+         - dest: 192.0.0.0/24
+           blackhole_config:
+             distance: 2
+           next_hops:
+             - forward_router_address: 192.1.1.11
+             - forward_router_address: 192.11.11.12
+             - forward_router_address: 192.3.3.13
+    state: replaced
+#
+#
+# -------------------------
+# Module Execution Result
+# -------------------------
+#
+#    "before": [
+#        {
+#            "address_families": [
+#                {
+#                    "afi": "ipv4",
+#                    "routes": [
+#                        {
+#                            "blackhole_config": {
+#                                "type": "blackhole"
+#                            },
+#                            "dest": "192.0.0.0/24",
+#                            "next_hops": [
+#                                {
+#                                    "forward_router_address": "192.11.11.11"
+#                                },
+#                                {
+#                                    "forward_router_address": "192.11.11.12"
+#                                }
+#                            ]
+#                        }
+#                    ]
+#                }
+#            ]
+#        },
+#        {
+#            "address_families": [
+#                {
+#                    "afi": "ipv6",
+#                    "routes": [
+#                        {
+#                            "blackhole_config": {
+#                                "distance": 2
+#                            },
+#                            "dest": "2001:db8::0/32",
+#                            "next_hops": [
+#                                {
+#                                    "forward_router_address": "2001:db8::1"
+#                                },
+#                                {
+#                                    "forward_router_address": "2001:db8::5"
+#                                }
+#                            ]
+#                        }
+#                    ]
+#                }
+#            ]
+#        }
+#    ]
+#
+# "commands": [
+#        "delete protocols static route 192.0.0.0/24 blackhole",
+#        "delete protocols static route 192.0.0.0/24 next-hop '192.11.11.11'",
```

Comment:
I see u are deleting `192.11.11.11` entry, but in the input `192.11.11.11` is not passed. So from replaced state logic ideally `192.11.11.11` shouldn't be deleted.

Reply:
yes you are right , so I have updated the same , thanks !!!